src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -72,34 +86,30 @@ private int getCostMultiplier(ItemStack tool) {
     public void addInformation(ItemStack stack, @Nullable World world, List<ITextComponent> tooltip, ITooltipFlag flag) {
         super.addInformation(stack, world, tooltip, flag);
         tooltip.add(TooltipTranslation.GADGET_DESTROYWARNING
-                            .componentTranslation()
-                            .setStyle(Styles.RED));
+                .componentTranslation()
+                .setStyle(Styles.RED));
         tooltip.add(TooltipTranslation.GADGET_DESTROYSHOWOVERLAY
-                            .componentTranslation(String.valueOf(getOverlay(stack)))
-                            .setStyle(Styles.AQUA));
+                .componentTranslation(String.valueOf(getOverlay(stack)))
+                .setStyle(Styles.AQUA));
         tooltip.add(TooltipTranslation.GADGET_BUILDING_PLACE_ATOP
-                            .componentTranslation(String.valueOf(getConnectedArea(stack)))
-                            .setStyle(Styles.YELLOW));
+                .componentTranslation(String.valueOf(getConnectedArea(stack)))
+                .setStyle(Styles.YELLOW));
         if (Config.isServerConfigLoaded() && Config.GADGETS.GADGET_DESTRUCTION.nonFuzzyEnabled.get())
             tooltip.add(TooltipTranslation.GADGET_FUZZY
-                                .componentTranslation(String.valueOf(getFuzzy(stack)))
-                                .setStyle(Styles.GOLD));
+                    .componentTranslation(String.valueOf(getFuzzy(stack)))
+                    .setStyle(Styles.GOLD));
 
         addInformationRayTraceFluid(tooltip, stack);
         addEnergyInformation(tooltip, stack);
     }
 
-    @Nullable
     public static String getUUID(ItemStack stack) {
-        NBTTagCompound tagCompound = stack.getTag();
-        if (tagCompound == null) {
-            tagCompound = new NBTTagCompound();
-        }
-        String uuid = tagCompound.getString(NBTKeys.GADGET_UUID);
+        NBTTagCompound tag = NBTHelper.getOrNewTag(stack);
+        String uuid = tag.getString(NBTKeys.GADGET_UUID);
```

Comment:
The Uuid should never have been a String in the first place. Could you migrate these to proper UUID uses?
 If it is too much to convert the worldsave to a UUID keyed Map, you can still call toString there